model/task/task.go

```diff
@@ -1385,14 +1385,19 @@ func (t *Task) SetGeneratedTasksToActivate(buildVariantName, taskName string) er
 func SetTasksScheduledTime(tasks []Task, scheduledTime time.Time) error {
 	ids := []string{}
 	for i := range tasks {
-		tasks[i].ScheduledTime = scheduledTime
-		ids = append(ids, tasks[i].Id)
+		// Skip tasks with scheduled time to prevent large updates
+		if utility.IsZeroTime(tasks[i].ScheduledTime) {
+			tasks[i].ScheduledTime = scheduledTime
+			ids = append(ids, tasks[i].Id)
+		}
 
 		// Display tasks are considered scheduled when their first exec task is scheduled
 		if tasks[i].IsPartOfDisplay() {
 			ids = append(ids, utility.FromStringPtr(tasks[i].DisplayTaskId))
 		}
 	}
+	// Remove duplicates to prevent large updates
+	ids = utility.UniqueStrings(ids)
```

Comment:
This is logically the same as #8377. The reason the query is complaining is because `utility.UniqueStrings` can return a nil slice if `ids` is empty, which translates to null in BSON and MQL requires a non-null array to be passed to `$in`. ([example playground](https://mongoplayground.net/p/peX_1gz5Cfm)

Generally, it's best to check if a slice is empty before passing it to `$in`. If it's empty, the query won't match anything anyways, so you can avoid running the query altogether.

Reply:
I was confused about why that gave an error so thank you for explaining that
I also added a test case with 0 updates!